### PR TITLE
[DB + API]: DB: created base structure of the project: needed liquibase files, also added docker and models. API: added swagger to the project

### PR DESCRIPTION
**Pull request by RostyslavKuzmych**

### CODE REVIEW

**Review by Andriy-Hrynevetskyi — APPROVED** [review]

File: .env [hunk 1]
```diff
@@ -0,0 +1,11 @@
...
+JWT.SECRET=fhjhdfkgjfkjgdkfjfhdfhgfjfgjfjfjgf
+ARRAY_OF_ENDPOINTS=/api/auth/**,/swagger-ui/**,/v3/api-docs/**
+MYSQLDB_USER=root
+MYSQLDB_ROOT_PASSWORD=myPassword
+MYSQLDB_DATABASE=task_management_db
+MYSQLDB_LOCAL_PORT=3308
+MYSQLDB_DOCKER_PORT=3306
+SPRING_LOCAL_PORT=8088
+SPRING_DOCKER_PORT=8080
+DEBUG_PORT=5005
```
Comment (AgroFix) [inline .env:11]: add empty line

File: docker-compose.yml [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+    environment:
+      - MYSQL_ROOT_PASSWORD=$MYSQLDB_ROOT_PASSWORD
+      - MYSQL_DATABASE=$MYSQLDB_DATABASE
+    ports:
+      - $MYSQLDB_LOCAL_PORT:$MYSQLDB_DOCKER_PORT
+  app:
+    depends_on:
+      - mysqldb
+    restart: on-failure
+    image: book_store
```
Comment (BolsunovaNataliia) [inline docker-compose.yml:18]: Need change name image.

File: src/main/java/application/security/JwtAuthenticationFilter.java [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+                    userDetails.getPassword(), userDetails.getAuthorities());
+            SecurityContextHolder.getContext().setAuthentication(authentication);
+        }
+        filterChain.doFilter(request, response);
+    }
+
+    private String getToken(HttpServletRequest request) {
+        String bearerToken = request.getHeader(AUTHORIZATION);
+        if (StringUtils.hasText(bearerToken) && bearerToken.startsWith(BEARER)) {
+            return bearerToken.substring(7);
```
Comment (AgroFix) [inline src/main/java/application/security/JwtAuthenticationFilter.java:50]:
```suggestion
            return bearerToken.substring(BEARER_PREFIX.length());

```
no magic numbers

File: .env (hunk 1 above)
Comment (Andriy-Hrynevetskyi) [inline .env:11]: empty line)

File: src/main/resources/db/changelog/changes/01-create-users-table.yaml [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+                  name: last_name
+                  type: varchar(255)
+                  constraints:
+                    nullable: false
+              - column:
+                  name: is_deleted
+                  type: boolean
+                  defaultValueBoolean: false
+                  constraints:
+                    nullable: false
```
Comment (Andriy-Hrynevetskyi) [inline src/main/resources/db/changelog/changes/01-create-users-table.yaml:47]: same here

File: src/main/resources/db/changelog/changes/08-add-roles-to-roles-table.yaml [hunk 5]
```diff
@@ -0,0 +1,23 @@
...
+                  value: ADMIN
+        - insert:
+            tableName: roles
+            columns:
+              - column:
+                  name: id
+                  value: 2
+              - column:
+                  name: role_name
+                  value: USER
```
Comment (Andriy-Hrynevetskyi) [inline src/main/resources/db/changelog/changes/08-add-roles-to-roles-table.yaml:23]: and here)

File: src/main/resources/db/changelog/changes/09-create-users_roles-table.yaml [hunk 6]
```diff
@@ -0,0 +1,24 @@
...
+                    primaryKey: true
+                    nullable: false
+              - column:
+                  name: role_id
+                  type: bigint
+                  constraints:
+                    foreignKeyName: fk_users_roles_role_id
+                    references: roles(id)
+                    primaryKey: true
+                    nullable: false
```
Comment (Andriy-Hrynevetskyi) [inline src/main/resources/db/changelog/changes/09-create-users_roles-table.yaml:24]: +

**Review by Andriy-Hrynevetskyi — CHANGES_REQUESTED** [review]
add some emty lines)

File: src/main/resources/db/changelog/changes/02-create-projects-table.yaml [hunk 7]
```diff
@@ -0,0 +1,46 @@
...
+        - createTable:
+            tableName: projects
+            columns:
+              - column:
+                  name: id
+                  type: bigint
+                  autoIncrement: true
+                  constraints:
+                    nullable: false
+                    primaryKey: true
```
Comment (BolsunovaNataliia) [inline src/main/resources/db/changelog/changes/02-create-projects-table.yaml:15]: nullable: false - redundant together with primaryKey: true

File: src/main/resources/db/changelog/changes/01-create-users-table.yaml [hunk 8]
```diff
@@ -0,0 +1,47 @@
...
+      changes:
+        - createTable:
+            tableName: users
+            columns:
+              - column:
+                  name: id
+                  type: bigint
+                  autoIncrement: true
+                  constraints:
+                    nullable: false
```
Comment (AgroFix) [inline src/main/resources/db/changelog/changes/01-create-users-table.yaml:14]: you dont need nullable: false, when you have primaryKey

**Review by BolsunovaNataliia — APPROVED** [review]

File: src/main/java/application/model/Attachment.java [hunk 9]
```diff
@@ -0,0 +1,38 @@
...
+@SQLDelete(sql = "UPDATE attachments SET is_deleted = true WHERE id = ?")
+@SQLRestriction(value = "is_deleted = false")
+@Data
+@Accessors(chain = true)
+@Table(name = "attachments")
+public class Attachment {
+    @Id
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
+    private Long id;
+    @ManyToOne
```
Comment (AgroFix) [inline src/main/java/application/model/Attachment.java:27]:
```suggestion
    @ManyToOne(fetch = FetchType.LAZY)

```

File: src/main/java/application/model/Comment.java [hunk 10]
```diff
@@ -0,0 +1,39 @@
...
+    @JoinColumn(nullable = false)
+    @ManyToOne
+    private Task task;
+    @ManyToOne
+    @JoinColumn(nullable = false)
+    private User user;
+    @Column(nullable = false)
+    private String text;
+    @Column(nullable = false)
+    private LocalDateTime timeStamp;
```
Comment (AgroFix) [inline src/main/java/application/model/Comment.java:36]:
```suggestion
    @Column(nullable = false)
       @CreationTimestamp
    private LocalDateTime timeStamp;

```

**Review by AgroFix — APPROVED** [review]

**Review by DEz-1996 — APPROVED** [review]

**Review by BolsunovaNataliia — APPROVED** [review]

**Review by AgroFix — APPROVED** [review]

**Review by Andriy-Hrynevetskyi — APPROVED** [review]

**Review by BolsunovaNataliia — APPROVED** [review]

**Review by AgroFix — APPROVED** [review]

**Review by Andriy-Hrynevetskyi — APPROVED** [review]

**Pull request merged**